### DISCUSSION TITLE
[Bug] Clerk 好像无法正常接入

### DISCUSSION THREAD

**Discussion by Yanyutin753** [lobehub/lobehub#2947] [Self Hosting] [answered]
### 💻 Operating System

Windows

### 📦 Environment

Docker

### 🌐 Browser

Edge

### 🐛 Bug Description

![image](https://github.com/lobehub/lobe-chat/assets/132346501/b2add276-bc6c-432f-b77a-92703631ca9b)

### 🚦 Expected Behavior

正常使用clerk

## 填入环境变量
![image](https://github.com/lobehub/lobe-chat/assets/132346501/da278843-53d0-4ea8-b924-70d2d9833649)

[a comment by an automated account is not shown]

**Comment by arvinxx** [comment, marked as answer]
因为用到了 NEXT_PUBLIC 的环境变量，可能需要你自行构建一个 Docker 镜像

**Reply by King-WKR** [reply]
大佬，向您请教一下，自行构建的话只需添加在Dockerfile这个文件里就好了嘛，我看了一下问题底下的其他大佬，似乎是修改在了Dockerfile.other这里，但是目前的项目文件里似乎并没有这个文件了。
cc: @Yanyutin753

**Reply by SpeedupMaster** [reply]
> 大佬，向您请教一下，自行构建的话只需添加在Dockerfile这个文件里就好了嘛，我看了一下问题底下的其他大佬，似乎是修改在了Dockerfile.other这里，但是目前的项目文件里似乎并没有这个文件了。 cc: @Yanyutin753

添加在Dockfile.database里

**Reply by King-WKR** [reply]
大佬，只需要ENV NEXT_PUBLIC_CLERK_PUBLISHABLE_KEY=pk_test_Y***********这个嘛
我看到手册上还有
CLERK_SECRET_KEY: sk_test_************************
CLERK_WEBHOOK_SECRET: w******************
这两个就不需要了嘛

**Reply by SpeedupMaster** [reply]
> 大佬，只需要ENV NEXT_PUBLIC_CLERK_PUBLISHABLE_KEY=pk_test_Y_**这个嘛 我看到手册上还有 CLERK_SECRET_KEY: sk_test_**_************* CLERK_WEBHOOK_SECRET: w****************** 这两个就不需要了嘛

```
ARG NEXT_PUBLIC_CLERK_PUBLISHABLE_KEY
ARG CLERK_SECRET_KEY
ARG CLERK_WEBHOOK_SECRET
ENV NEXT_PUBLIC_CLERK_PUBLISHABLE_KEY="pk_live_Y2xlcmxxxxxxxx" \
    CLERK_SECRET_KEY="" \
    CLERK_WEBHOOK_SECRET=""
```
我前几天添加的这些构建的

**Reply by King-WKR** [reply]
> > 大佬，只需要ENV NEXT_PUBLIC_CLERK_PUBLISHABLE_KEY=pk_test_Y_**这个嘛 我看到手册上还有 CLERK_SECRET_KEY: sk_test_**_************* CLERK_WEBHOOK_SECRET: w****************** 这两个就不需要了嘛
> 
> ```
> ARG NEXT_PUBLIC_CLERK_PUBLISHABLE_KEY
> ARG CLERK_SECRET_KEY
> ARG CLERK_WEBHOOK_SECRET
> ENV NEXT_PUBLIC_CLERK_PUBLISHABLE_KEY="pk_live_Y2xlcmxxxxxxxx" \
>     CLERK_SECRET_KEY="" \
>     CLERK_WEBHOOK_SECRET=""
> ```
> 
> 我前几天添加的这些构建的

感谢！

[a comment by an automated account is not shown]

**Comment by Yanyutin753** [comment]
佬有办法合并吗🥺

[a comment by an automated account is not shown]

**Comment by arvinxx** [comment]
@Yanyutin753 我之前看有一些方法似乎可以做到 Docker 里也能启动时注入 NEXT_PUBLIC 变量，但是我没有试过，目前也没有精力搞这个（不是主线）。 所以可能需要社区的小伙伴来帮忙看看了

[a comment by an automated account is not shown]

**Comment by Yanyutin753** [comment]
好 我明天看看能不能加进去 我也记得可以的 前两天还加了next_public

[a comment by an automated account is not shown]

**Comment by Yanyutin753** [comment]
@arvinxx 佬早点休息！

[a comment by an automated account is not shown]

**Comment by Yanyutin753** [comment]
通过在dockerfile里面加上环境变量即可使用，但是作者大佬，有个问题，在网不是很好的情况下会出现这个情况，有机会修护吗！
![image](https://github.com/lobehub/lobe-chat/assets/132346501/4cd85d67-6f9a-42b3-80fc-0278a23db490)

**Comment by arvinxx** [comment]
@Yanyutin753 网络不好大概率不行吧。因为需要连到Clerk他们的服务器的

[a comment by an automated account is not shown]

**Comment by yinmao** [comment]
@Yanyutin753 请问Dockerfile中环境变量如何添加的呢？我添加在CMD ["node", "server.js"]上面，如下：`ENV NEXT_PUBLIC_CLERK_PUBLISHABLE_KEY "pk_live_xxx"
ENV CLERK_SECRET_KEY "sk_live_xxx"
CMD ["node", "server.js"]`
build后仍然报相同错误

[a comment by an automated account is not shown]

**Comment by ponypy** [comment]
@Yanyutin753 请问怎么在dockerfile中添加啊，我有点不懂，可以教一下我吗

[a comment by an automated account is not shown]

**Comment by Yanyutin753** [comment]
https://github.com/Yanyutin753/lobe-chat/blob/main/Dockerfile.other 替换就行

**Comment by ponypy** [comment]
> https://github.com/Yanyutin753/lobe-chat/blob/main/Dockerfile.other 替换就行

我是用的docker部署，怎么才能找到这个文件呢

**Comment by Yanyutin753** [comment]
> > https://github.com/Yanyutin753/lobe-chat/blob/main/Dockerfile.other 替换就行
> 
> 
> 
> 我是用的docker部署，怎么才能找到这个文件呢

自己打包

**Comment by Matinal111** [comment]
@arvinxx @Yanyutin753 我通过分支main打包会出现这个错误，麻了
```
ERROR: failed to solve: process "/bin/sh -c pnpm i" did not complete successfully: exit code: 1
```

**Reply by Yanyutin753** [reply]
https://github.com/Yanyutin753/lobe-chat/blob/main/Dockerfile.other 试试这个
